editor/src/clj/editor/properties.clj

```diff
@@ -460,15 +460,29 @@
         new-value (from-fn new-value)]
     (set-value-txs evaluation-context node-id prop-kw key set-fn value new-value)))
 
-(defn- set-values [evaluation-context property values]
+(defn- resolve-set-operations
+  "Resolves the supplied property and sequence of new values to a vector of
+  individual [node-id prop-kw old-value new-value] vectors detailing the
+  low-level property set operations that will be performed as a result."
+  [property values]
+  (let [node-ids (:node-ids property)
+        prop-kws (:prop-kws property)
+        old-values (:values property)
+        new-values (if-some [from-fn (-> property :edit-type :from-type)]
+                     (mapv from-fn values)
+                     values)]
+    (mapv vector node-ids prop-kws old-values new-values)))
```

Comment:
Refactoring: A `property` map in the Property Editor can refer to one or more properties on various nodes. For example, when I select a `EmbeddedComponentNode` representing a Sprite component in the Property Editor, it is also presenting the properties from the embedded `SpriteNode`.

We need this sequence of resolved "set-operations" in `set-values` below, but also to calculate the `edited-endpoints`.